### PR TITLE
feat: add forms for login and register

### PR DESCRIPTION
**Pull request by perkinson1251**
### Migrated from React project:
- SignIn form;
- SignUp form;

### Errors:
1. `isDisabled` doesn't work, but I don't know why. Code:
```ts
const isDisabled = [emailValue.value, passwordValue.value].some((val) => val === '')
```

2. Submit don't work, but I don't know why. Error screen:
chrome_MG3UMCelLE

### CODE REVIEW

File: src/core/composables/useSigninForm.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { getFormData, handleFieldError } from '@/core/utils'
+import { flatten, parse } from 'valibot'
+import { authSchema } from '../schemas/auth.schema'
+
+export const useSigninForm = () => {
+  const { signin, isLoading, error } = useSignin()
+  const emailValue = ref('')
+  const passwordValue = ref('')
+  const validationError = ref<IValidationError | null>(null)
+  const isDisabled = ref([emailValue.value, passwordValue.value].some((val) => val === ''))
```
Comment (OneLiL05) [inline src/core/composables/useSigninForm.ts:13]: I guess, here you need to change `ref` to `computed`, the code will look like this:

```ts
const isDisabled = computed(() => {
  return [emailValue.value, passwordValue.value].some((val) => val === '')
})
```

File: src/core/composables/useSigninForm.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  const handleEmailChange = (e: any) => {
```
Comment (OneLiL05) [inline src/core/composables/useSigninForm.ts:15]: Remove this function since it's useless

File: src/core/composables/useSigninForm.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const value = e.target.value
+    emailValue.value = value
+  }
+
+  const handlePasswordChange = (e: any) => {
```
Comment (OneLiL05) [inline src/core/composables/useSigninForm.ts:20]: Remove this function

File: src/core/composables/useSignupForm.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import { getFormData, handleFieldError } from '@/core/utils'
+import { flatten, parse } from 'valibot'
+import { authSchema } from '../schemas/auth.schema'
+
+export const useSignupForm = () => {
+  const { signup, isLoading, error } = useSignup()
+  const emailValue = ref('')
+  const passwordValue = ref('')
+  const validationError = ref<IValidationError | null>(null)
+  const isDisabled = ref([emailValue.value, passwordValue.value].some((val) => val === ''))
```
Comment (OneLiL05) [inline src/core/composables/useSignupForm.ts:13]: Change `ref` to the `computed`

File: src/core/composables/useSignupForm.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  const handleEmailChange = (e: any) => {
```
Comment (OneLiL05) [inline src/core/composables/useSignupForm.ts:15]: Remove this function

File: src/core/composables/useSignupForm.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    const value = e.target.value
+    emailValue.value = value
+  }
+
+  const handlePasswordChange = (e: any) => {
```
Comment (OneLiL05) [inline src/core/composables/useSignupForm.ts:20]: Remove this function

File: src/components/ui/TextField/TextField.vue [hunk 7]
```diff
@@ -1,18 +1,19 @@
...
-  modelValue: string
-  error?: string | null
+  value: string
+  error?: IValidationError | string | null
 }
 const props = defineProps<TextFieldProps>()
 
-const emit = defineEmits(['update:modelValue'])
-const model = useVModel(props, 'modelValue', emit)
+const emit = defineEmits(['update:value'])
```
Comment (OneLiL05) [inline src/components/ui/TextField/TextField.vue:15]: Change it to the previous version

File: src/components/ui/TextField/TextField.vue [hunk 8]
```diff
@@ -1,18 +1,19 @@
...
+const model = useVModel(props, 'value', emit)
```
Comment (OneLiL05) [inline src/components/ui/TextField/TextField.vue:16]: Change it to the previous version

**Review by OneLiL05 — CHANGES_REQUESTED** [review]
Read comment, and resolve issues

**Comment by perkinson1251** [comment]
New commit has one major bug: buttons `Увійти` and `Зареєструватися` don't work, but I don't know why

File: src/router/index.ts [hunk 9]
```diff
@@ -42,15 +44,15 @@ const router = createRouter({
...
+      component: LoginView
     },
     {
       path: '/signup',
       meta: {
         title: 'Реєстрація'
       },
       name: 'signup',
-      component: () => import('../views/AboutView.vue')
+      component: SignupView
```
Comment (OneLiL05) [inline src/router/index.ts:55]: Change to `() => import('../views/SignupView.vue')`

File: src/router/index.ts [hunk 10]
```diff
@@ -42,15 +44,15 @@ const router = createRouter({
         title: 'Вхід'
       },
       name: 'login',
-      component: () => import('../views/AboutView.vue')
+      component: LoginView
```
Comment (OneLiL05) [inline src/router/index.ts:47]: Change to `() => import('../views/LoginView.vue')`

File: src/layouts/AuthLayout/AuthLayout.vue [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  </div>
+</template>
+<style lang="scss" scoped>
+.Layout {
+  @apply flex flex-col justify-center items-center w-full h-[100dvh];
+}
+.LayoutContainer {
+  @apply flex flex-col justify-center items-center bg-transparent gap-5;
+
+  @media (min-width: 900px) {
```
Comment (OneLiL05) [inline src/layouts/AuthLayout/AuthLayout.vue:25]: Change to `min-width: theme('screens.md')`

**Review by OneLiL05 — CHANGES_REQUESTED** [review]
There are both `tailwindcss` issues and two unresolved issues from previous review, resolve all of them and pr will be ready to be merged

**Comment by perkinson1251** [comment]
A new bug has appeared: the theme does not work on the `Login` and `Signup` pages, what I mean is that you can choose a `Dark` theme, but on these pages the theme will be chosen randomly. And the bugs with the `Login` and `Register` buttons are not fixed.

We should meet in Discord today, because I don't really understand how to fix these bugs

**Review by OneLiL05 — APPROVED** [review]

**Pull request merged**